test/typescript/asset/test.ts

```diff
@@ -125,3 +126,115 @@ describe("full integration test", () => {
     expect(await driver.diff("normal")).toContain("No changes");
   });
 });
+
+describe("createAssetsFromLocalModules test", () => {
+  test("remote source without createAssetsFromLocalModules", () => {
+    const remoteSource = "terraform-aws-modules/eks/aws";
+
+    const app = Testing.stubVersion(new App({ stackTraces: false }));
+    const stack = new TerraformStack(app, "MyStack");
+
+    const moduleOptionsNotExists = new TerraformHclModule(
+      stack,
+      "moduleOptionsNotExists",
+      {
+        source: remoteSource,
+      }
+    );
+
+    expect(moduleOptionsNotExists.source).toEqual(remoteSource);
+  });
+
+  test("remote source with createAssetsFromLocalModules set to true", () => {
+    const remoteSource = "terraform-aws-modules/eks/aws";
+
+    const app = Testing.stubVersion(new App({ stackTraces: false }));
+    const stack = new TerraformStack(app, "MyStack");
+
+    const moduleOptionsTrue = new TerraformHclModule(
+      stack,
+      "moduleOptionsTrue",
+      {
+        source: remoteSource,
+        createAssetsFromLocalModules: true,
+      }
+    );
+
+    expect(moduleOptionsTrue.source).toEqual(remoteSource);
+  });
+
+  test("remote source with createAssetsFromLocalModules set to false", () => {
+    const remoteSource = "terraform-aws-modules/eks/aws";
+
+    const app = Testing.stubVersion(new App({ stackTraces: false }));
+    const stack = new TerraformStack(app, "MyStack");
+
+    const moduleOptionsFalse = new TerraformHclModule(
+      stack,
+      "moduleOptionsFalse",
+      {
+        source: remoteSource,
+        createAssetsFromLocalModules: false,
+      }
+    );
+
+    expect(moduleOptionsFalse.source).toEqual(remoteSource);
+  });
+
+  test("local source without createAssetsFromLocalModules", () => {
+    const localSource = "../";
+    const assetRegex =
+      /^.\/assets\/MyStack_localmoduleid_[a-zA-z\d]{8}\/[a-zA-z\d]+$/;
+
+    const app = Testing.stubVersion(new App({ stackTraces: false }));
+    const stack = new TerraformStack(app, "MyStack");
+
+    const moduleOptionsNotExists = new TerraformHclModule(
+      stack,
+      "moduleOptionsNotExists",
+      {
+        source: localSource,
+      }
+    );
+
+    expect(moduleOptionsNotExists.source).toMatch(assetRegex);
+  });
+
+  test("local source with createAssetsFromLocalModules set to true", () => {
+    const localSource = "../";
+    const assetRegex =
+      /^.\/assets\/MyStack_localmoduleid_[a-zA-z\d]{8}\/[a-zA-z\d]+$/;
```

Comment:
This is a bit unfortunate, but I think our overrides to make asset testing easier aren't exposed for module assets.

Reply:
Apologizes. Just meant that it's unfortunate that we don't have the ability to make module created asses use the same simplified test output as direct assets. No need for changes at this time though.